TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PID.java

```diff
@@ -0,0 +1,38 @@
+package org.firstinspires.ftc.teamcode;
+
+import com.qualcomm.robotcore.hardware.PIDCoefficients;
+
+public class PID {
+
+    private PIDCoefficients pid;
+    private double target;
+    private double lastError = 0;
+    private double integral = 0;
+
+    public PID(double k_p, double k_i, double k_d) {
+        pid = new PIDCoefficients(k_p, k_i, k_d);
+    }
+
+    //sets the target of the pid
+    public void setTarget(double in) {

```

Comment:
you didn't consider the start position of the robot. I will recommend to change this:
```suggestion
    public void resetTargets(double target, double startPosition, double startTime) {

```
and call this function before every time you start calling the outputForce function

Reply:
there is no point in reseting the target because it is different for every class that uses the PID

Reply:
the start position of the robot is true however

Reply:
The starting position of the robot is included in the "input" of outputForce, the start time also does not matter all that matters is the delta time.

Reply:
Therefore what @godboi524 has is fine

Reply:
no no. you're incorrect about the startTime, because I need the how much time passed for the certain runtime of a function, not the entire robot. And as to the startPosition, there's no point. Say we have the robot, we want it to move a certain distance, the start position will be the distance to the target anyway. If we have a certain speed we want to get to, I guess the speed of the motor could be higher than 0, but that would be done outside of the PID class so that there won't be a start position for distance.

Reply:
Actually no, you wouldn't even need a start speed for the motor. We calculate the difference anyway.

Reply:
do what you think is ok, just make sure the PID calculate is correct at the end